lib/spack/spack/util/compression.py

```diff
@@ -14,45 +14,45 @@
 # Supported archive extensions.
 PRE_EXTS   = ["tar", "TAR"]
 EXTS       = ["gz", "bz2", "xz", "Z"]
-NOTAR_EXTS = ["zip", "tgz", "tbz", "tbz2", "txz"]
+NOTAR_EXTS = ["zip", "tgz", "tbz2", "tbz", "txz"]
 
 # Add PRE_EXTS and EXTS last so that .tar.gz is matched *before* .tar or .gz
 ALLOWED_ARCHIVE_TYPES = [".".join(ext) for ext in product(
     PRE_EXTS, EXTS)] + PRE_EXTS + EXTS + NOTAR_EXTS
 
 is_windows = sys.platform == 'win32'
 
+try:
+    import bz2 # noqa
+    _bz2_support = True
+except ImportError:
+    _bz2_support = False
 
-def bz2_support():
-    try:
-        import bz2 # noqa
-        return True
-    except ImportError:
-        return False
```

Comment:
You are correct that the primary motivation was to avoid performing a large number of imports. Specifically, the flow of calling `tar_support()` which would attempt to import `tarfile` immediately followed by the line `import tarfile` on return of a `True` value. This felt congested and wasteful. I did consider it initially but I don't believe that memoizing is the right approach, as that would still retain the function scope for the imported modules, precluding general module usage without retuning the module directly from the method. Unless I misunderstood, if so, I can revert this change.
I wouldn't call it essential to the PR, or any functionality.

Reply:
This is fine, my main goal here was to understand which parts of the PR are essential.